src/main/java/me/rigamortis/seppuku/impl/module/misc/HotBarRefillModule.java

```diff
@@ -0,0 +1,185 @@
+     * @return The index of the largest stack of the given item, -1 if the given item does not exist.
+     */
+    public int getBiggestStack (EntityPlayerSP player, ItemStack item, boolean includeHotbar) {
+        if (item == null) {
+            return -1;
+        }
+
+        int maxCount = -1;
+        int maxIndex = -1;
+        for (int i = includeHotbar ? 0 : 9; i < player.inventory.mainInventory.size(); i++) {
+            ItemStack stack = player.inventory.mainInventory.get(i);
+
+            if (stack.getItem() != Items.AIR
+                && stack.getItem() == item.getItem()
+                && stack.getCount() > maxCount) {
+
+                maxCount = stack.getCount();
+                maxIndex = i;
+            }
+        }
+
+        return maxIndex;
+    }
+
+    /**
+     * Refills a given slot in the hotbar from an item in the player's inventory.
+     * Uses the slot's current ItemStack to decide what it should be refilled with.
+     *
+     * @param mc The passed Mincraft instance
+     * @param slot The slot that should be refilled
+     */
+    public void refillHotbarSlot(Minecraft mc, int slot) {
+        ItemStack stack;
+        if (slot == 45) { // Special case for offhand
+            stack = mc.player.getHeldItemOffhand();
+        } else {
+            stack = mc.player.inventory.mainInventory.get(slot);
+        }
+
+        // If the slot is air it cant be refilled
+        if (stack.getItem() == Items.AIR) {
+            return;
+        }
+
+        // The slot can't be refilled if there is nothing to refill it with
+        int biggestStack = getBiggestStack(mc.player, stack, false);
+        if (biggestStack == -1) {
+            return;
+        }
+
+        // Special case for offhand (can't use QUICK_CLICK)
+        if (slot == 45) {
+            mc.playerController.windowClick(mc.player.inventoryContainer.windowId, biggestStack, 0, ClickType.PICKUP, mc.player);
+            mc.playerController.windowClick(mc.player.inventoryContainer.windowId, 45, 0, ClickType.PICKUP, mc.player);
+            mc.playerController.windowClick(mc.player.inventoryContainer.windowId, biggestStack, 0, ClickType.PICKUP, mc.player);
+            return;
+        }
+
+        int overflow = -1; // The slot a shift click will overflow to
+        for (int i = 0; i < 9 && overflow == -1; i++) {
+            if (mc.player.inventory.mainInventory.get(i).getItem() == Items.AIR) {
+                overflow = i;
+            }
+        }
+
+
+        if (overflow == -1) { // If the hotbar is full we can just shift click biggestStack to refill slot
+            mc.playerController.windowClick(mc.player.inventoryContainer.windowId, biggestStack, 0, ClickType.QUICK_MOVE, mc.player);
+        } else { // If the hotbar isn't full, we might have to click the overflow stack back into the inventory
+            mc.playerController.windowClick(mc.player.inventoryContainer.windowId, biggestStack, 0, ClickType.QUICK_MOVE, mc.player);
```

Comment:
Is this if statement necessary? The lines are the same.

Reply:
It's necessary but he just has it in the wrong way.  He should move the windowClick(...) up before the if-statement and then flip the condition.  That will remove the duplicate line and correctly handle returning overflow items.

Reply:
I had to change something here earlier and just forgot to change it. Fixed.